util/src/main/java/io/kubernetes/client/util/WebSocketCall.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// This has been cloned from the okhttp-ws package so that I could remove
+// the requirement that websockets use the "GET" method
```

Comment:
I think we should fix that too but we should not wait for that. I will investigate it (to see why it gets into the spec and if we actually support POST) but I don't think there is anything we can do with POST that we can't do with GET. Websocket reuse the connection and ignore the method, thus I think we should not support POST in java client (or any other client).